lib/internal/modules/esm/module_job.js

```diff
@@ -295,22 +295,33 @@ class ModuleJobSync extends ModuleJobBase {
   #loader = null;
   constructor(loader, url, importAttributes, moduleWrap, isMain, inspectBrk) {
     super(url, importAttributes, moduleWrap, isMain, inspectBrk, true);
-    assert(this.module instanceof ModuleWrap);
     this.#loader = loader;
-    const moduleRequests = this.module.getModuleRequests();
-    // Specifiers should be aligned with the moduleRequests array in order.
-    const specifiers = Array(moduleRequests.length);
-    const modules = Array(moduleRequests.length);
-    const jobs = Array(moduleRequests.length);
-    for (let i = 0; i < moduleRequests.length; ++i) {
-      const { specifier, attributes } = moduleRequests[i];
-      const job = this.#loader.getModuleJobForRequire(specifier, url, attributes);
-      specifiers[i] = specifier;
-      modules[i] = job.module;
-      jobs[i] = job;
+
+    assert(this.module instanceof ModuleWrap);
```

Comment:
Can we add a message to this assert? And should we keep this assert in Release?

Reply:
It should be in release mode, the whole point of it is to defend against bugs and ask users to open an issue if it gets hit.

Reply:
I don't know if there can be a better message than..."this.module must be an instance of ModuleWrap, otherwise this is a bug and users should open an issue to https://github.com/nodejs/node/issues", which is already what the message will say for `internal/assert`.